aristanetworks/ansible-avd
# Feat(eos_designs): Allow 'default' in default_interfaces node types

## Change Summary

Allow the use of 'default' in `default_interfaces.*.types` key.

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Specifying 'default' in the list of types will apply these default interfaces to all node types (the node's platform must still match a platform type regular expression).

No change to the existing data model, other than documenting the use of the 'all' keyword in the schema.

```
default_interfaces:
  - types: [ default ]
    platforms: [ 7300X3 ]
    uplink_interfaces: [ Ethernet1/27-30/1 ]
    mlag_interfaces: [ Ethernet1/31-32/1 ]
    downlink_interfaces: [ Ethernet1/1-26/1 ]
```

## How to test
Existing molecule tests modified to include `types: [ default ]` similar to above.

Create a default_interface entry as per example above - should apply to all 7300X devices, regardless of node type.

## Checklist

### Repository Checklist
